lib/barbeque/message_handler/job_execution.rb

```diff
@@ -6,19 +6,21 @@ module Barbeque
   module MessageHandler
     class JobExecution
       # @param [Barbeque::Message::JobExecution] message
-      # @param [Barbeque::JobQueue] job_queue
-      def initialize(message:, job_queue:)
+      # @param [Barbeque::MessageQueue] message_queue
+      def initialize(message:, message_queue:)
         @message = message
-        @job_queue = job_queue
+        @message_queue = message_queue
       end
 
       def run
         begin
-          job_execution = Barbeque::JobExecution.create(message_id: @message.id, job_definition: job_definition, job_queue: @job_queue)
+          job_execution = Barbeque::JobExecution.create(message_id: @message.id, job_definition: job_definition, job_queue: @message_queue.job_queue)
         rescue ActiveRecord::RecordNotUnique => e
+          @message_queue.delete_message(@message)
```

Comment:
I realized I had misunderstood about the duplication after reading some SQS documents. I skipped DeleteMessage
https://github.com/cookpad/barbeque/pull/43/commits/a3974145c0e3f03539078ad280c6dab7ab4b0551